### ISSUE TITLE
jQuery Resizable Containment does not work

### ISSUE THREAD

**Issue by thenajsays** [RWAP/jquery-ui-touch-punch#38]
I have a web application I built using jQuery Draggable and Resizable that I added jQuery Touch Punch to for touch screen functionality and the containment functionality for Resizable does not work.

**Comment by RWAP** [comment]
It is always difficult to comment on something like this - sample code or the web application for testing is of course necessary.

It may well be the order in which you load javascript, how the HTML page is built or simply the versions of jquery and jquery-ui you use.  It could even be the device you are using!

What exactly does not work?

Remember that all touch-punch does is to try to mimic mouse buttons / moves by interpreting finger touches / movements.

**Comment by thenajsays** [comment]
It does not constrain the resizable element in the parent container. I am using jQuery 3.6.0 and jQuery UI 1.13.2.
Files should load in this order:
1. jQuery
2. jQuery UI
3. jQuery touch punch
4. my scripts

Here is a link to the application: https://tinyurl.com/5xmykmjw

To test the application:

1. Select a frame size and an image (any but the first, as that is a pattern will not have the resizable functionality)
2. Click configure
3. Drag the frame and drag it
4. Use the drag handle to resize the frame, you will see the it extends past the image boundaries. 
This can be tested both on a touch screen or a desktop, the functionality works the same either way. If I remove touch punch it works exactly as expected on desktop.

Of note, to partially remedy the issue, I dynamically set the max height and width for the resizable element to match the height and width of the image, but the only remedies situations where the user resizes the element while it is in the top left corner.

**Comment by RWAP** [comment]
I can replicate the issue here on a PC using your test site - in Firefox (desktop view), if I move a box to the middle, I can extend it beyond the boundary of the image using the mouse

It does not appear to be limited to touch screens.

**Comment by RWAP** [comment]
On a side note, your test page still uses touch-punch v0.2.2 which is woefully out of date

**Comment by thenajsays** [comment]
>It does not appear to be limited to touch screens.

That is correct. If I disable touch punch it works correctly on desktop. 


>On a side note, your test page still uses touch-punch v0.2.2 which is woefully out of date

I will update and report back

**Comment by RWAP** [comment]
Which script has code to set up the draggable containment as per https://stackoverflow.com/questions/1547662/setting-boundary-limits-for-a-draggable-object

**Comment by thenajsays** [comment]
> Which script has code to set up the draggable containment as per https://stackoverflow.com/questions/1547662/setting-boundary-limits-for-a-draggable-object

scenic-panel-scripts.js

**Comment by RWAP** [comment]
Actually from a programming point of view, I wonder if the issue is more to do with that every time the #configure-panels button is clicked, you set up:
```

				$('img#panel-img-panel-frame').resizable({
					aspectRatio: panelIMG_W / panelIMG_H,
					containment: '#panel-img-config-panel-component',
					maxHeight: (IMG_H * IMG_ratio),
					maxWidth: baseWidth
				});

```
You may need to destroy this and re-assert it if it already exists.

**Comment by thenajsays** [comment]
I have a destroy function set up and just added a fallback. I actually have the latest version of touch punch set up to load but I think a plugin I'm using may be loading the older version, superseding mine. I will have to look into this further. I will report back with results.

**Comment by thenajsays** [comment]
Ok, that was the culprit. Apparently WordPress has that old version of your plugin built in, so when I attempted to enqueue the new version, it defaulted to theirs because the ID I assigned my theme copy matched the ID they assigned to theirs. I reregistered the correct version and it works beautifully now. thank you for helping me troubleshoot. It never would have occurred to me that WordPress had it as part of their default library if you hadn't pointed out the version number!

I will be reaching out to WordPress to advise them that they need to update their library!
